scraper.rb

```diff
@@ -32,29 +26,14 @@ def datefrom(date)
   url = URI.join(BASE, page).to_s
   warn "Fetching #{url}"
 
-  page = noko(url)
-
-  page.css('div.view-members-of-parliament div.panel-display').each do |entry|
-    mp_url = URI.join(BASE, entry.css('.views-field-view-node a/@href').text).to_s
-
-    mp = noko(mp_url)
-
-    party = entry.css('span.views-field-field-political-party .field-content').text.strip
-    (party_name, party_id) = party.match(/(.*) \((.*)\)/).captures
+  MembersPage.new(response: Scraped::Request.new(url: url).response)
+             .member_rows
+             .each do |row|
+    mp_page = MemberPage.new(response: Scraped::Request.new(url: row.source).response)
```

Comment:
The staggered indentation makes this quite difficult to read. I'd suggest using the `scrape` helper method that we use in other places, so this can be simplified:

```ruby
def scrape(h)
  url, klass = h.to_a.first
  klass.new(response: Scraped::Request.new(url: url).response)
end
```